File: Kode/obliger/oblig2/python/Movie.py

class Movie:

    def __init__(self, id, title, rating, votes):
        self.id = id
        self.title = title
        self.rating = rating
        self.votes = votes
        self.actors = {}

    def addActor(self, id, actor):
        self.actors[id] = actor

    def getActors(self):
        return self.actors

    def getRating(self):
        return float(self.rating)

    def __str__(self):
        return str(self.title)+" ("+str(self.rating)+")"

    def __repr__(self):
        return str(self.title)

File: Kode/obliger/oblig2/java/Movie.java
package obliger.oblig2;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Movie {

    String id, title, votes;
    Double rating;
    List<Actor> playingIn = new ArrayList<>();

    public Movie(String id, String title, Double rating, String votes) {
        this.id = id;
        this.title = title;
        this.rating = rating;
        this.votes = votes;
    }

    public void addActor(Actor a) {
        playingIn.add(a);
    }

    public boolean hasActor(Actor a) {
        return playingIn.contains(a);
    }

    // getters
    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getVotes() {
        return votes;
    }

    public Double getRating() {
        return rating;
    }

    public List<Actor> getActors() { return playingIn; }

    @Override
    public String toString() {
        return title+"\n"+playingIn;
    }
}
